http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/sam/SamAuthProvider.scala

```diff
@@ -152,6 +152,21 @@ class SamAuthProvider[F[_]: Effect: Logger](samDao: SamDAO[F],
     checkPersistentDiskPermissionWithProjectFallback(samResource, authorization, action, googleProject)
   }
 
+  def getRuntimeActions(samResource: SamResource, userInfo: UserInfo)(
+    implicit ev: ApplicativeAsk[F, TraceId]
+  ): F[List[RuntimeAction]] = {
+
+    val authorization = Authorization(Credentials.Token(AuthScheme.Bearer, userInfo.accessToken.token))
+
+    for {
+      listOfPermissions <- samDao
+        .getListOfResourcePermissions(samResource, authorization)
+
+    } yield {
+      listOfPermissions.map(x => runtimeActionMap.find(_._2 == x).map(_._1).get)
```

Comment:
This `.get` doesn't seem ideal.. and this probably isn't efficient since it's traversing `runtimeActionMap` each time.

What if we declared this in the class:
```
private val runtimeActionMapReversed = runtimeActionMap.map(_.swap)
```
Then here we could just yield:
```
listOfPermissions.flatMap(runtimeActionMapReversed.get)
```

Reply:
@xuscrbw I wonder if this is where that `None.get` error you were seeing is coming from :)